app/pages/notifications.cjsx

```diff
@@ -0,0 +1,145 @@
+React = require 'react'
+Loading = require '../components/loading-indicator'
+Paginator = require '../talk/lib/paginator'
+ChangeListener = require '../components/change-listener'
+talkClient = require '../api/talk'
+Notification = require './notifications/notification'
+
+module?.exports = React.createClass
+  displayName: 'NotificationsPage'
+
+  propTypes:
+    project: React.PropTypes.object
+    user: React.PropTypes.object
+
+  getDefaultProps: ->
+    query: page: 1
+
+  getInitialState: ->
+    firstMeta: { }
+    lastMeta: { }
+    notificationsMap: { }
+
+  componentWillMount: ->
+    @getNotifications() if @props.user
+
+  componentWillUnmount: ->
```

Comment:
Mark the current top and bottom page of notifications as read when leaving this page.

This solves the problem of "reading" the first and last page of notifications